_integration-test/run.sh

```diff
@@ -173,23 +173,32 @@ echo "${_endgroup}"
 echo "Creating test replay..."
+
 TEST_REPLAY_ID=$(
   export LC_ALL=C
   head /dev/urandom | tr -dc "a-f0-9" | head -c 32
 )
 TIME_IN_SECONDS=$(date +%s)
-curl -sf --data '{"event_id":"'"$TEST_REPLAY_ID"'","sdk":{"name":"sentry.javascript.browser","version":"7.38.0"}}
+curl -sf --data '{"event_id":"'"$TEST_REPLAY_ID"'","sdk":{"name":"sentry.javascript.react","version":"7.31.1"}}
 {"type":"replay_event"}
-{"type":"replay_event","replay_start_timestamp":$TIME_IN_SECONDS,"timestamp":$TIME_IN_SECONDS,"error_ids":[],"trace_ids":[],"urls":["example.com"],"replay_id":"'"$TEST_REPLAY_ID"'","segment_id":0,"replay_type":"session","event_id":"'"$TEST_REPLAY_ID"'","environment":"production","sdk":{"name":"sentry.javascript.browser","version":"7.38.0"},"request":{"url":"example.com","headers":{"platform":"javascript","contexts":{"replay":{"session_sample_rate":1,"error_sample_rate":1}}}
+{"type":"replay_event","replay_start_timestamp":"'"$TIME_IN_SECONDS"'","timestamp":"'"$TIME_IN_SECONDS"'","error_ids":[],"trace_ids":[],"urls":["http://example.com/"],"replay_id":"'"$TEST_REPLAY_ID"'","segment_id":0,"replay_type":"session","event_id":"'"$TEST_REPLAY_ID"'","environment":"production","sdk":{"integrations":["InboundFilters","FunctionToString","TryCatch","Breadcrumbs","GlobalHandlers","LinkedErrors","Dedupe","HttpContext","Replay"],"version":"7.31.1","name":"sentry.javascript.react"},"sdkProcessingMetadata":{},"request":{"url":"http://example.com/","headers":{"User-Agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"}},"platform":"javascript","tags":{"sessionSampleRate":1,"errorSampleRate":1}}
 {"type":"replay_recording","length":19}
 {"segment_id":0}
 []' -H 'Content-Type: application/json' -H "X-Sentry-Auth: Sentry sentry_version=7, sentry_key=$SENTRY_KEY, sentry_client=test-bash/0.1" "$SENTRY_TEST_HOST/api/$PROJECT_ID/envelope/" -o /dev/null
-
 printf "Getting the test replay back"
-REPLAY_SEGMENT_PATH="api/0/projects/sentry/internal/replays/$TEST_EVENT_ID/recording-segments/?download"
-REPLAY_EVENT_PATH="api/0/projects/sentry/internal/replays/$TEST_EVENT_ID/"
-timeout 60 bash -c 'until $(sentry_api_request "$REPLAY_EVENT_PATH" -Isf -X GET -o /dev/null); do printf '.'; sleep 0.5; done'
-timeout 60 bash -c 'until $(sentry_api_request "$REPLAY_SEGMENT_PATH" -Isf -X GET -o /dev/null); do printf '.'; sleep 0.5; done'
+REPLAY_SEGMENT_PATH="api/0/projects/sentry/internal/replays/$TEST_REPLAY_ID/recording-segments/?download"
+REPLAY_EVENT_PATH="api/0/projects/sentry/internal/replays/$TEST_REPLAY_ID/"
+
+for i in {1..30}; do
+  REPLAY_EVENT_PROCESSED=$(sentry_api_request "$REPLAY_EVENT_PATH" -X GET | jq -r '.data.id')
+  REPLAY_RECORDING_PROCESSED=$(sentry_api_request "$REPLAY_SEGMENT_PATH" -X GET | jq .)
+  if ! [ $REPLAY_EVENT_PROCESSED = $TEST_REPLAY_ID ] || [ -z "$REPLAY_RECORDING_PROCESSED" ]; then
+    echo $i
+    sleep $i
+  else
+    break
+  fi
```

Comment:
The code this replaced clearly showed the intention with the `timeout 60` part. With this one, all I see is a magic number, `30` and increased magical sleeping.

1. Why do we need to wait longer between retries?
2. Please clearly document the reasoning and the working logic of the code with comments.

Reply:
1. this was a bug, thank you for the catch. 
2. added a couple of comments